db/migrate/20210322225751_create_reminder_templates.rb

```diff
@@ -0,0 +1,11 @@
+class CreateReminderTemplates < ActiveRecord::Migration[5.2]
+  def change
+    create_table :reminder_templates do |t|
+      t.integer :experiment_group, null: false
+      t.string :message, null: true
```

Comment:
I think we probably want to enforce `null: false` here...it wouldn't make sense to create a template without a message.

Reply:
The reason I left message nullable is perhaps a code smell. I was initially going to create an experiment_group (bucket) model that would belong_to reminder_experiment and have many reminder_templates. But it would have contained no data of its own. At most it would need an integer identifier starting at zero for use with modulo operations in the bucketing process. It would just be a join table between experiment and reminder template. This felt like an unnecessary abstraction, so I decided to include experiment_group on reminder_template that can be used for bucketing. 

Because an experiment group can have multiple message templates, that means message templates can look like this:
```
MessageTemplate
experiment_group: 0
message: appointment in 3 days
appointment_offset: -3

MessageTemplate
experiment_group: 0
message: you're late
appointment_offset: 3


MessageTemplate
experiment_group: 1
message: nil
appointment_offset: nil
```

The two things to note are that 
1) there can be multiple reminder_templates with the same experiment group
2) there is a message template representing the control group, which gets no message. And as I type this all out, I feel like I'm convincing myself that perhaps I should make ExperimentGroup a first class citizen. One of the reasons I chose to leave it out is that it felt over-architected to have a an ExperimentGroup that, at most, has an integer identifier to use for matching with modulos. And also because there will be a very complex query at the end of this project for fetching all of the data and I didn't see a good reason to add yet another join.

Reply:
> The reason I left message nullable is perhaps a code smell. I was initially going to create an experiment_group (bucket) model that would belong_to reminder_experiment and have many reminder_templates. But it would have contained no data of its own.

I think it does capture an important piece of the model though - it captures the intersection of:

* patients
* an experiment
* the reminders to be sent for that experiment

The patients we can capture via some sort of Flipper mechanism (we can discuss that in a sync today), but for Flipper to have a handle on the group we want some notion of that group.  So I think it makes sense to make the ExperimentGroup a first class model.

Then the null: true code smell here goes away.